mycore-base/src/main/java/org/mycore/datamodel/metadata/MCRObjectService.java

```diff
@@ -840,4 +927,371 @@ public final int getFlagIndex(String value) {
+        List<MCRMetaDateLangText> internalList = getMessagesAsMCRMetaLangText(type);
+        messages.removeAll(internalList);
+    }
+
+    /**
+     * This method return the size of the classification list.
+     *
+     * @return the size of the classification list
+     */
+    public final int getClassificationsSize() {
+        return this.classifications.size();
+    }
+
+    /**
+     * Returns the classifications as list.
+     *
+     * @return classifications as list
+     */
+    protected final List<MCRMetaClassification> getClassificationsAsList() {
+        return classifications;
+    }
+
+    /**
+     * This method returns all classification values of the specified type.
+     *
+     * @param classId
+     *              a classId as string.
+     * @return a list of classification values
+     */
+    protected final List<MCRMetaClassification> getClassificationsAsMCRMetaClassification(String classId) {
+        return classifications.stream()
+            .filter(metaClassification -> classId.equals(metaClassification.getClassId()))
+            .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * This method returns all classification values.
+     *
+     * @return a list of classification values
+     */
+    public final List<MCRCategoryID> getClassificationsAsMCRCategoryID() {
+        return getClassificationsAsList().stream()
+            .map(c -> new MCRCategoryID(c.getClassId(), c.getCategId()))
+            .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * This method returns all classification values of the specified type.
+     *
+     * @param classId
+     *              a classId as string.
+     * @return a list of classification values
+     */
+    public final List<String> getClassifications(String classId) {
+        return getClassificationsAsMCRMetaClassification(classId).stream()
+            .map(MCRMetaClassification::getCategId)
+            .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * This method returns all classification values of the specified type.
+     *
+     * @param classId
+     *              a classId as string.
+     * @return a list of classification values
+     */
+    public final List<MCRCategoryID> getClassificationsAsMCRCategoryID(String classId) {
+        return getClassificationsAsMCRMetaClassification(classId).stream()
+            .map(c -> new MCRCategoryID(c.getClassId(), c.getCategId()))
+            .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * This method get a single classification from the classification list as a string.
+     *
+     * @exception IndexOutOfBoundsException
+     *                throw this exception, if the index is invalid
+     * @return a classification string
+     */
+    public final String getClassification(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= classifications.size()) {
```

Comment:
always return MCRCategoryID for classification (categid as String is 'incomplete')

Reply:
I've actually raised this as a talking point (access to classifiactions vs. access to other multi-property-entities in the service section, like flags) in my mail to the developer mailing list this Monday. I'd prefer some kind of consistency. I hope we can discuss this in the next web conference.